tools/unitctl/unit-client-rs/src/unitd_docker.rs

```diff
@@ -0,0 +1,471 @@
+                ctr.container_image,
+                ctr.rewrite_socket(
+                    cmd.strip_prefix("/usr/local/bin/docker-entrypoint.sh")
+                        .or_else(|| Some(""))
+                        .unwrap()
+                        .to_string()
+                )
+            ))
+        });
+        let mut cmds = vec![];
+        let _ = command.map_or((), |cmd| cmds.push(cmd));
+        UnitdProcess {
+            all_cmds: cmds,
+            binary_name: ctr.container_image.clone(),
+            process_id: ctr
+                .details
+                .as_ref()
+                .and_then(|details| {
+                    details
+                        .state
+                        .as_ref()
+                        .and_then(|state| state.pid.and_then(|pid| Some(pid.clone() as u64)))
+                })
+                .or(Some(0 as u64))
+                .unwrap(),
+            executable_path: None,
+            environ: vec![],
+            working_dir: ctr.details.as_ref().and_then(|details| {
+                details.config.as_ref().and_then(|conf| {
+                    Some(
+                        PathBuf::from(
+                            conf.working_dir
+                                .as_ref()
+                                .map_or(String::new(), |dir| ctr.host_path(dir.clone())),
+                        )
+                        .into_boxed_path(),
+                    )
+                })
+            }),
+            child_pids: vec![],
+            user: None,
+            effective_user: None,
+            container: Some(ctr.clone()),
+        }
+    }
+}
+
+impl Serialize for UnitdContainer {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        // 5 = fields to serialize
+        let mut state = serializer.serialize_map(Some(5))?;
+        state.serialize_entry("container_id", &self.container_id)?;
+        state.serialize_entry("container_image", &self.container_image)?;
+        state.serialize_entry("command", &self.command)?;
+        state.serialize_entry("mounts", &self.mounts)?;
+        state.serialize_entry("platform", &self.platform)?;
+        state.end()
+    }
+}
+
+impl UnitdContainer {
+    pub async fn find_unitd_containers() -> Vec<UnitdContainer> {
+        if let Ok(docker) = Docker::connect_with_local_defaults() {
+            match docker.list_containers::<String>(None).await {
+                Err(e) => {
+                    eprintln!("{}", e);
+                    vec![]
+                }
+                Ok(summary) => {
+                    // cant do this functionally because of the async call
+                    let mut mapped = vec![];
+                    for ctr in summary {
+                        if ctr.clone().image
+                            .or(Some(String::new()))
+                            .unwrap()
+                            .contains("unit") {
```

Comment:
This will match all Docker containers that contain "unit", which might match something like `united/foobar`.